File: gradle-plugins/gradletestapps/java-one-project/src/main/java/com/example/execution/javaoneproject/resolvers/GreetingResolver.java
package com.example.execution.javaoneproject.resolvers;

import com.example.execution.javaoneproject.resolvers.resolverbases.QueryResolvers;
import java.util.concurrent.CompletableFuture;
import viaduct.java.api.annotations.Resolver;

@Resolver
public class GreetingResolver extends QueryResolvers.Greeting {
  @Override
  public CompletableFuture<String> resolve(Context ctx) {
    return CompletableFuture.completedFuture("hello from java-one-project");
  }
}

File: gradle-plugins/gradletestapps/multi-project/alpha/src/main/kotlin/com/example/execution/multiproject/alpha/GreetingResolver.kt
package com.example.execution.multiproject.alpha

import com.example.execution.multiproject.alpha.resolverbases.QueryResolvers
import viaduct.api.resolver.Resolver

@Resolver(queryValueFragment = "fragment _ on Query { secretGreeting }")
class GreetingResolver : QueryResolvers.Greeting() {
    override suspend fun resolve(ctx: Context) = "hello from multi-project alpha"
}

@Resolver
class SecretGreetingResolver : QueryResolvers.SecretGreeting() {
    override suspend fun resolve(ctx: Context) = "private"
}
